src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const totalArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = Object.assign(stateCopy, action.extraData);
```

Comment:
```suggestion
        stateCopy = { ...stateCopy, ...action.extraData };
```
it can be shorter with spread operator